### DISCUSSION TITLE
module federation load it's chunks from a wrong url

### DISCUSSION THREAD

**Discussion by joebnb** [webpack/webpack#18422] [Q&A] [answered]
# Bug report

**What is the current behavior?**

## remote app
when using module federation webpack will generate:
* entry ` eg:static/remoteEntry.js `
* entry's dependenceies `eg:static/remoteEntry_index.ad3ds.js` 

this remote app this app running in `localhost:10000`

```js
 new ModuleFederationPlugin({
      name: 'remoteEntry',
      library: { type: 'window', name: 'remoteEntry' },
      filename: 'static/remoteEntry.js',
      exposes: {
        './entry': './src/remoteEntry',
      },
    }),
```

## host app
configurate remoteEntry.js address `eg:http://localhost:10000/static/remoteEntry.js` in host app.

config like:
```js
  new ModuleFederationPlugin({
      name: 'host',
      remotes: {
        remoteEntry:'portalRemote@http://localhost:10000/static/remoteEntry.js'
      },
    }),
```

visit host app `http://localhost:10001` **NOTIC:remote and host are different origin**

chrome console will print remoteEntry's chunks will 404 beacause it's will load from `http://localhost:10001/remoteEntry_index.ad3ds.js`
**request were use incorrect origin and lost pathname**

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

actually expect if i config host app like `http://localhost:10000/static/remoteEntry.js`,module federation plugin will load it's dependence from `http://localhost:10000/static/` not `http://localhost:10001`,

if specified remote url, the remote artifacts should read from remote, host app may not have remote app files.

if this is not a bug, would give me some suggestion or explain how to workaround this

-------
screenshots

![image](https://github.com/webpack/webpack/assets/1646883/4ac98d2c-75c1-4937-82fa-2cc814c0ffbd)

![image](https://github.com/webpack/webpack/assets/1646883/bb41deb4-c581-4f4f-86df-80149e5f743c)

**Comment by alexander-akait** [comment]
Please provide reproducible test repo using github, I will show how to fix it

**Reply by joebnb** [reply, marked as answer]
i solved it by using publicpath `auto`

**Comment by lateshkudnekar** [comment]
setting publicpath to auto is not working for me, Is there any other option ?

**Reply by leolima** [reply]
same here
